10/umbraco-deploy/handling-cache-refresher-notifications.md

```diff
@@ -0,0 +1,162 @@
+---
+meta.Title: "Handling Cache Refresher Notifications"
+description: "How to respond to deployment events using cache refresher notifications"
+---
+
+# Handling Cache Refresher Notifications
+
+## Background
+
+When you deploy content or other Umbraco data between environments, some notifications that are normally fired in CMS operations are suppressed.
+
+For example, you may handle a `ContentPublishedNotification` to apply some custom logic when a content item is published. This code will run in a normal CMS publish operation. However, when deploying a content item into another environment and triggering it's publishing there, the notification will not be issued. And the custom logic in the notification handler will not run.
+
+This behavior is deliberate and done for performance and reliability reasons. A normal save and publish operation by an editor operates on one item at a time. With deployments, we may have many, and publishing these notifications may lead to at best slow operations, and at worst inconsistent data.
+
+However, what if you do want to run some code on an update, even if this is happening as part of a Deploy operation?
+
+There's an option here using cache refresher notifications. Not all events are suppressed by Umbraco Deploy. Some that are batched up and fired after the deploy operation is completed include those related to refreshing the Umbraco cache.
+
+## Implementing a Cache Refresher Notification
+
+The following two code samples illustrate how this can be done.
+
+The first handles a content cache refresher, which takes a payload from where the content ID can be extracted. Using that the content can be retrieved in the local Umbraco instance and used as appropriate.
```

Comment:
Is it possible to make it even clearer in which scenarios the ContentCacheRefresherNotificationHandler will be executed. Will it execute on all content nodes, only the updated ones, only the saved ones?

Reply:
I think all I can really say here is that Deploy will issue all cache refresher notifications that are issued by the CMS.  It doesn't suppress any of these.  Probably further detail would need to be in the CMS documentation if necessary, but I'll make sure that we at least make this key point: that you'll get the same notifications issued as if the operation was carried out by an editor in the backoffice of the environment that's being deployed to.